R/knitr-engine.R

```diff
@@ -628,3 +642,54 @@ eng_python_autoprint <- function(captured, options, autoshow) {
   }
 
 }
+
+save_python_session <- function(cache_path) {
+  module <- tryCatch(import("dill"), error = identity)
+  if (inherits(module, "error")) {
+    if (module$message == "ImportError: No module named dill") return()
+    signalCondition(module$message)
+  }
+
+  r_obj_exists <- "'r' in globals()"
+  r_is_R <- "type(r).__module__ == '__main__' and type(r).__name__ == 'R'"
+  if (py_eval(r_obj_exists) && py_eval(r_is_R)) {
```

Comment:
Rather than evaluating this code as a string, you should use `reticulate` APIs for manipulating objects.

Reply:
The R object is now ignored using `dill`'s new variable filter.